src/types/redis_hash.h

```diff
@@ -63,9 +65,16 @@ class Hash : public SubKeyScanner {
                        std::vector<std::string> *values = nullptr);
   rocksdb::Status RandField(const Slice &user_key, int64_t command_count, std::vector<FieldValue> *field_values,
                             HashFetchType type = HashFetchType::kOnlyKey);
+  rocksdb::Status ExpireFields(const Slice &user_key, uint64_t expire_ms, const std::vector<Slice> &fields, 
+                               HashFieldExpireType type, bool is_persist, std::vector<int8_t> *ret);
+  rocksdb::Status TTLFields(const Slice &user_key, const std::vector<Slice> &fields, std::vector<int64_t> *ret);
+  static bool IsExpiredField(Metadata &metadata, const Slice &value);
 
  private:
   rocksdb::Status GetMetadata(Database::GetOptions get_options, const Slice &ns_key, HashMetadata *metadata);
+  static rocksdb::Status decodeFieldValue(const HashMetadata &metadata, std::string *value, uint64_t &expire);
+  static rocksdb::Status encodeValueExpire(std::string *value, uint64_t expire);
+  static bool isMeetCondition(HashFieldExpireType type, uint64_t new_expire, uint64_t old_expire);
```

Comment:
Can we change the name of this function to describe more clearly what it checks? What question does it answer?

Reply:
Agree with what @torwig mentioned, for:

`decodeFieldAndTTL` should be separated into two methods: `decodeExpireFromValue` and `checkIfFieldExpired`, and `encodeFieldAndTTL` can rename to `encodeExpireToValue`. @torwig What do you think?